src/sensors.py

```diff
@@ -79,24 +82,37 @@ def get_updates():
 
 # Temperature method depending on system distro
 def get_temp():
-    temp = ''
-    if 'rasp' in OS_DATA['ID']:
-        reading = subprocess.check_output(['vcgencmd', 'measure_temp']).decode('UTF-8')
-        temp = str(re.findall('\d+.\d+', reading)[0])
-    else:
-        reading = subprocess.check_output(['cat', '/sys/class/thermal/thermal_zone0/temp']).decode('UTF-8')
-        temp = str(reading[0] + reading[1] + '.' + reading[2]) # why?? need linux system to test
+    temp = 'Unknown'
+    # Utilising psutil for temp reading on ARM arch
+    try:
+        temp = psutil.sensors_temperatures()['cpu_thermal'][0].current
+    except:
+        try:
+            # Assumes that first entry is the CPU package, have not tested this on other systems except my NUC x86
+            temp = psutil.sensors_temperatures()['coretemp'][0].current
+        except Exception as e:
+            print('Could not establish CPU temperature reading: ' + str(e))
+            raise
```

Comment:
I was debating doing away with a seperate call for Raspi, but the comments over in [this SO thread](https://raspberrypi.stackexchange.com/questions/41784/temperature-differences-between-cpu-gpu) suggest that `thermal_zone` is an alias of `vcgencmd` on the Pi and that the latter is the most direct way of getting the data. I still don't think the performance increase will be anywhere near perceivable, so using a single command would be better and far more maintainable.
While I can't verify this, [this issue](https://github.com/j7126/OctoPrint-Dashboard/issues/148) reports an as-of-yet unfixed bug with `psutil` temperature readings when 1wire sensors are connected via gpio. I would be in favour of getting the required data in the most direct way possible to avoid errors caused by dependencies; at least wherever it's not a hassle to reinvent the wheel.